generative_data_prep/utils/__init__.py

```diff
@@ -24,10 +24,18 @@
     TokenTypeIds,
 )
 from .large_file_shuffle import large_file_shuffle
+from .logger import (
+    add_file_handler,
+    log_current_datetime,
+    log_elapsed_time,
+    log_git_commit_hash,
+    log_input_args,
+    log_metrics,
+    log_sep_str,
+)
```

Comment:
i don't know why we do this; this is so unnecessary; people should be importing modules with explicit paths rather than doing this gymnastics.

Reply:
this is within the same module, so this is standard practice

Reply:
For example, look at PyTorch: https://github.com/pytorch/pytorch/blob/main/torch/__init__.py#L26

Reply:
I don't know what I'm missing but they're simply importing libraries so that people can get around importing them without explicit paths. These imports are actually being used in the code:

https://github.com/pytorch/pytorch/blob/main/torch/__init__.py#L634

Reply:
so everything we import here is used by other modules in our codebase.    The __init__ is used to define which functions are publicly accessible to other modules

Reply:
I'm missing something here; where is this documented in Python? I understand we can define `__all__` to specify which modules are available by default which I don't see here. Besides that, what other purpose does this serve? If functions are not supposed to be consumed by end-users, shouldn't we be naming them with `_name`, `__name`, and `__name__` accordingly?

Reply:
Here is another example: https://github.com/pytorch/pytorch/blob/main/torch/utils/data/__init__.py

Reply:
I'm still confused what the concern is haha

Reply:
ok, this is being used in `__all__`; totally missed that. Ok, scratch this comment.

Reply:
ohh I see, got it, yeah the __all__ was hidden away haha